source/server/utils.cc

```diff
@@ -34,8 +34,13 @@ void assertExclusiveLogFormatMethod(
 
 void maybeSetApplicationLogFormat(
     const envoy::config::bootstrap::v3::Bootstrap::ApplicationLogConfig& application_log_config) {
-  if (application_log_config.has_log_format() &&
-      application_log_config.log_format().has_json_format()) {
+  if (!application_log_config.has_log_format()) {
+    return;
+  }
+
+  if (application_log_config.log_format().has_text_format()) {
+    Logger::Registry::setLogFormat(application_log_config.log_format().text_format());
```

Comment:
What happens if someone sets the CLI option? I think this is similar to the other convo we had? Should we have an error if someone sets the CLI option and then sets the bootstrap option? I admit that perhaps there is a use case where someone could set the log format before the bootstrap is read, but that seems like kind of an edge case and we could just see if anyone complains about that? wdyt?

Reply:
I agree - if we decided to follow some pattern then I think it should be applied to all cases.
The exclusive format method is done before stepping into this function, in:
https://github.com/envoyproxy/envoy/blob/72c2b962244585cb782161a732eea46f88cc6838/source/server/server.cc#L426

So this case should be covered, and any other log format that would be added in the future.

Reply:
OK thanks